util/include/SimpleMemoryPool.hpp

```diff
@@ -51,6 +51,17 @@ class SimpleMemoryPool {
     }
   }
 
+  virtual ~SimpleMemoryPool() {
+    if (freeCallback_) {
```

Comment:
what happens if there is no freeCallback_? is it even possible?

Reply:
yes, it's possible if the user decided not to define it (and not define allocCallback). We don't check if one is defined and another one is not, it's the user's responsibility.
You can see examples where it happen in concord-bft/util/test/simple_memory_pool_test.cpp
It can happen if the pool manages some kind of native types or fixed-size objects. It never happens in ST, which is currently the only user of this util (as far as I know) and needs allocation and free callbacks.

Reply:
looking forward isn't it bug prone? allowing the user to leak memory if they don't correctly define the callback?